These variables are all static local so declaring them as 
extern in their header file is pointless and just breaks.

--- plugin/key/key_large/key_large.h
+++ plugin/key/key_large/key_large.h
@@ -43,7 +43,7 @@
 
 typedef union key_large key_large_t;
 
-extern reiser4_plug_t key_large_plug;
+static reiser4_plug_t key_large_plug;
 
 typedef enum {
 	/* Major locality occupies higher 60 bits of the first element */
--- plugin/key/key_short/key_short.h
+++ plugin/key/key_short/key_short.h
@@ -42,7 +42,7 @@
 
 typedef union key_short key_short_t;
 
-extern reiser4_plug_t key_short_plug;
+static reiser4_plug_t key_short_plug;
 
 typedef enum {
 	/* Major locality occupies higher 60 bits of the first element */
--- plugin/oid/oid40/oid40.h
+++ plugin/oid/oid40/oid40.h
@@ -30,7 +30,7 @@
 
 typedef struct oid40 oid40_t;
 
-extern reiser4_plug_t oid40_plug;
+static reiser4_plug_t oid40_plug;
 
 #define oid40_get_next(area)            LE64_TO_CPU(*((d64_t *)area))
 #define oid40_set_next(area, val)       (*((d64_t *)area) = CPU_TO_LE64(val))
